src/lib/login.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 Adobe Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+const { getToken, invalidateToken, context } = require('@adobe/aio-lib-ims')
+const { CLI } = require('@adobe/aio-lib-ims/src/context')
+const aioLogger = require('@adobe/aio-lib-core-logging')('@adobe/aio-cli-plugin-app-templates:lib-login', { provider: 'debug' })
+
+/**
+ * Login to AIO CLI and generate new access token
+ *
+ * @returns {Promise<string>} access token
+ */
+async function retrieveAccessToken () {
+  const current = await context.getCurrent()
```

Comment:
This plugin will be used by `aio cli`, so I believe we can use the code from https://github.com/adobe/aio-cli-plugin-events/blob/ad539a5b72d48c14d78b43ee3c89149508d97ec9/src/BaseCommand.js#L30-L33 , for example.